### ISSUE TITLE
[Bug]: No notifiaction mails are sent if mentioned in file comments

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/server#32936]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

No notification is sent when a comment on a file is posted with a personal notification.

In the activity (web frontend) the comment is mentioned. Thus, it is detected correctly by the server. Only the immediate sending of mails is not done.

The delayed sending of the mails is not affected. Other mails can also be sent. So, the problem is not related to the mailing system.

### Steps to reproduce

1. Set the personal notifications to send mails on comments on files. 
![image](https://user-images.githubusercontent.com/8202922/174480164-62059ee5-54f2-4de4-9cad-7acc1ed07329.png)
2. Also, set the notifications for email reminders for unhandled notifications
![image](https://user-images.githubusercontent.com/8202922/174480179-7f828d8f-e969-48fc-8e08-950ae6394262.png)
3. Prepare another account **A** and add a shared account.
4. Put a comment as user **A** on the shared file with a mention to the main account (with `@...`).
5. Have a look at the mails now and in one day

### Expected behavior

The mail is immediately sent out when Nextcloud is configured to send out the mail upon comments.

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.tsc-vfl.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.1.1",
        "overwrite.cli.url": "https:\/\/cloud.tsc-vfl.de",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "app_install_overwrite": [
            "twofactor_admin",
            "files_texteditor"
        ],
        "maintenance": false,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtpauthtype": "PLAIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - appointments: 1.12.6
  - apporder: 0.15.0
  - bruteforcesettings: 2.4.0
  - calendar: 3.3.2
  - checksum: 1.1.4
  - circles: 24.0.0
  - cloud_federation_api: 1.7.0
  - collectives: 1.2.0
  - comments: 1.14.0
  - contacts: 4.1.1
  - contactsinteraction: 1.5.0
  - dav: 1.22.0
  - deck: 1.7.1
  - external: 4.0.0
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_antivirus: 3.3.0
  - files_downloadactivity: 1.13.0
  - files_linkeditor: 1.1.10
  - files_markdown: 2.3.6
  - files_mindmap: 0.0.26
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_texteditor: 2.14.0
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - forms: 2.5.1
  - groupfolders: 12.0.0
  - groupquota: 0.1.8
  - impersonate: 1.11.0
  - integration_google: 1.0.6
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - mail: 1.13.4
  - metadata: 0.16.0
  - nextcloud_announcements: 1.13.0
  - notes: 4.3.1
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - oidc_login: 2.3.2
  - password_policy: 1.14.0
  - photos: 1.6.0
  - polls: 3.6.1
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - quota_warning: 1.14.0
  - recommendations: 1.3.0
  - richdocuments: 6.1.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - sociallogin: 4.15.3
  - spreed: 14.0.2
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - transfer: 0.5.1
  - twofactor_backupcodes: 1.13.0
  - twofactor_email: 2.4.0
  - twofactor_nextcloud_notification: 3.4.0
  - twofactor_totp: 6.4.0
  - twofactor_webauthn: 0.3.1
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - user_usage_report: 1.8.0
  - viewer: 1.8.0
  - workflowengine: 2.6.0
Disabled:
  - admin_audit
  - dashboard: 7.4.0
  - encryption
  - files_external
  - twofactor_u2f
  - user_ldap
  - weather_status: 1.4.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Additional info

I am using the official docker image under an Archlinux host.

I am unsure if this problem is related to the NC server or to a dedicated app. I asked for help [on the forum](https://help.nextcloud.com/t/mail-notifications-not-sent-out-immeadiately-for-comments-on-files/140162) but did not get any answer so far.

I tried to search for this problem but did not find anything useful so far. If it is in fact a duplicate, please be kind and direct me to the correct location.

Here are some information extracted from the server directly: https://cloud.tsc-vfl.de/index.php/s/DGM3FqPRaQmWEJy

Regarding the Nextcloud logs, at what point in time do you need the logs. At the time of the mentioning (missing mail) or later? Currently, the file is roughly 100MB in size and I cannot create a gist from it for some reason.

**Comment by CarlSchwan** [comment]
Couldn't find anything interesting in the logs but you might want to fix your file antivirus config ;)

**Comment by christianlupus** [comment]
> Couldn't find anything interesting in the logs but you might want to fix your file antivirus config ;)

Yeah, the antivirus should be fixed by now :smile:. Thanks for looking at this.

**Comment by szaimen** [comment]
This rather sounds like a feature request to me

**Comment by christianlupus** [comment]
Well, I am with you, @szaimen in general. This is more like a feature request. However, then the last setting confuses me:

![grafik](https://user-images.githubusercontent.com/8202922/174480164-62059ee5-54f2-4de4-9cad-7acc1ed07329.png)

What is meant by this exactly? When would I get mail notifications (if the checkbox is set) w.r.t. to comments? What situation would that be?

**Comment by LukasFreeze** [comment]
It would be great if any comment would trigger a notificiation (as the checkboxes shown by @christianlupus suggest), not just if you are mentioned.

Currently, even if I actively go to `/apps/activity/?filter=comments`, only my own comments are shown which is basically useless in our cloud which is intended for cooperation with several users.

**Comment by valerio-bozzolan** [comment]
Thanks again for this reported issue. I cannot reproduce your exact issue anymore. Can you?

Since I'm affected by a slightly different thing, I describe this relevant use-case:

When a person shares a comment on a file, everybody with file visibility receive an email notification (subject `Activity at Nextcloud XYZ`).

Some users want to disable such notifications, and they want to be notified only when they are involved through an explicit mention in the comment (`@username`). But this seems not possible. Because, at the moment, we only have this option:

> Comments for files
> [X] Mail
> [X] Push

Screenshot:

<img width="936" height="315" alt="Nextcloud screenshot of Notification settings for Comments for File" src="https://github.com/user-attachments/assets/4c4ce458-5d00-4a22-9a33-9014acaa1a19" />

So:

- if you have "Comments for files" enabled, you always receive comment notifications (with or without mention).
- if you have "Comments for files" disabled, nobody receives notifications, even with explicit mention!

In short:

Who is explicitly mentioned should always be notified somehow.

As solution, an extra option "Comments for file when mentioned" could be added. Checked as default.
